internal/jobworker/job/job.go

```diff
@@ -0,0 +1,302 @@
+package job
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/tjper/teleport/internal/jobworker"
+	"github.com/tjper/teleport/internal/jobworker/output"
+	"github.com/tjper/teleport/internal/jobworker/reexec"
+
+	"github.com/google/uuid"
+)
+
+// New creates a new Job instance.
+func New(
```

Comment:
Looking back at this, I'd roll job.Job's API entirely into JobService. Callers have to go through multiple hoops to get actions done, where we could have a simpler API

For example:

* Starting a job is job.New followed by jobService.StartJob (skipping the JobService creation). There's nothing useful to do with that job pointer before StartJob.
* Streaming is jobService.FetchJob followed by job.StreamOutput
* Status is jobService.FetchJob followed by job.Status
* Stopping is just jobService.StopJob, which I think is better, but is an outlier compared to others.